### PR TITLE
DEX-208 add annotation to Encounter linkage plus patching of annotation from one encounter to another

### PR DESCRIPTION
**Pull request by hwindsor**
Title says it all

## Pull Request Overview

Annotation to Encounter linkage and patching added

---

**Database Updates [OPTIONAL]**
Annotation to encounter many to one mapping added

**REST API Updates [OPTIONAL]**

Annotation creation now requires an encounter_guid as well as an asset guid

Patch of Annotation added to change encounter
**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly covered
  - Example: The percentage of the code covered by tests has not decreased
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

File: app/modules/annotations/parameters.py [hunk 1]
```diff
@@ -5,18 +5,47 @@
...
+    @classmethod
+    def replace(cls, obj, field, value, state):
+        from app.modules.encounters.models import Encounter
+
+        ret_val = False
+        # Annotations don't have an owner, so check the encounter owner
+        if (
+            rules.owner_or_privileged(current_user, obj.encounter)
+            or current_user.is_admin
+        ):
```
Comment (colinwkingen) [inline app/modules/annotations/parameters.py:44]: WBIA might cause some move actions like this post MVP without a user being (directly) involved, but we can cross that bridge later.

File: app/requirements.txt [hunk 2]
```diff
@@ -48,7 +48,7 @@ scikit-build
 sentry-sdk>=0.10.2
 sentry-sdk[flask]>=0.10.2
 
-setuptools>=42
+setuptools>=45
```
Comment (colinwkingen) [inline app/requirements.txt:51]: Did an error prompt this or just keeping current?

**Review by colinwkingen — APPROVED** [review]
LGTM, this is a pretty key ability to have.

File: app/requirements.txt (hunk 2 above)
Comment (hwindsor) [reply app/requirements.txt:51]: An error, I ended up pulling in a versions of setuptools_scm (below) that required setuptools 45 and the above line was getting 44 and it all failed.

**Pull request merged**